### PR TITLE
Make various EXIT, the well's MAX and some leftover buttons translatable and fix the castle EXIT button

### LINKED ISSUE #7461: Castle screen, EXIT button has wrong background in a pressed state
### Preliminary checks

- [X] I've checked that there aren't [**other open issues**](https://github.com/ihhub/fheroes2/issues?q=is%3Aissue) on the same topic.
- [X] I've checked that this issue is reproducible on the [**latest snapshot build**](https://ihhub.github.io/fheroes2/INSTALL.html#snapshots-latest-builds).

### Platform

Windows

### Describe the bug

When pressing EXIT button we can see brown background line above it.

image

### Save file

None.

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by zenseii**
This fixes the issues with EXIT buttons in the Puzzle, Dimension Door, Heroes Meeting, Town screens, and translates buttons found in buying a boat dialog, and the MAX button in the Well.

Fix #7461
Relates to #7496

Part of #6105

### CODE REVIEW

**Review by Districh-ru — APPROVED** [review]
Well done, @zenseii!

изображение 
изображение 
изображение изображение

File: src/fheroes2/agg/agg_image.cpp [hunk 1]
```diff
@@ -917,20 +920,80 @@ namespace fheroes2
...
+                Point releasedOffset = { 7, 5 };
+                Point pressedOffset = { 6, 6 };
+                renderTextOnButton( _icnVsSprite[id][0], _icnVsSprite[id][1], gettext_noop( "smallerButton|EXIT" ), releasedOffset, pressedOffset,
+                                    { textWidth, fheroes2::getFontHeight( fheroes2::FontSize::BUTTON_RELEASED ) }, fheroes2::FontColor::WHITE );
+
+                break;
+            }
+            case ICN::BUTTON_EXIT_PUZZLE_DDOOR_GOOD:
+            case ICN::BUTTON_EXIT_PUZZLE_DDOOR_EVIL: {
+                _icnVsSprite[id].resize( 2 );
```
Comment (ihhub) [inline src/fheroes2/agg/agg_image.cpp:970]: Please cache **_icnVsSprite[id]** for better performance and readability.

File: src/fheroes2/agg/agg_image.cpp [hunk 2]
```diff
@@ -917,20 +920,80 @@ namespace fheroes2
...
+                    // clean the button.
+                    Fill( out, 6 - i, 4 + i, 71 - i, 17, getButtonFillingColor( i == 0, !isEvilInterface ) );
+                }
+
+                const int32_t textWidth = 71;
+                Point releasedOffset = { 6, 5 };
+                Point pressedOffset = { 5, 6 };
+                renderTextOnButton( _icnVsSprite[id][0], _icnVsSprite[id][1], gettext_noop( "smallerButton|EXIT" ), releasedOffset, pressedOffset,
+                                    { textWidth, fheroes2::getFontHeight( fheroes2::FontSize::BUTTON_RELEASED ) },
+                                    isEvilInterface ? fheroes2::FontColor::GRAY : fheroes2::FontColor::WHITE );
```
Comment (ihhub) [inline src/fheroes2/agg/agg_image.cpp:996]: Please either make **releasedOffset** and **pressedOffset** const (you can remove `=`) or just directly use { 6, 5 } and { 5, 6 } in the call below.

File: src/fheroes2/agg/agg_image.cpp [hunk 3]
```diff
@@ -917,20 +920,80 @@ namespace fheroes2
...
+                    // clean the button.
+                    Fill( out, 6 - i, 4 + i, 70, 17, getButtonFillingColor( i == 0 ) );
+                }
+
                 const int32_t textWidth = 70;
-                createNormalButton( _icnVsSprite[id][0], _icnVsSprite[id][1], textWidth, gettext_noop( "EXIT" ), isEvilInterface );
+                Point releasedOffset = { 7, 5 };
+                Point pressedOffset = { 6, 6 };
+                renderTextOnButton( _icnVsSprite[id][0], _icnVsSprite[id][1], gettext_noop( "smallerButton|EXIT" ), releasedOffset, pressedOffset,
+                                    { textWidth, fheroes2::getFontHeight( fheroes2::FontSize::BUTTON_RELEASED ) }, fheroes2::FontColor::WHITE );
```
Comment (ihhub) [inline src/fheroes2/agg/agg_image.cpp:964]: Please either make **releasedOffset** and **pressedOffset** const (you can remove `=`) or just directly use { 7, 5 } and { 6, 6 } in the call below.

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @zenseii , I left few comments here. Could you please check them?

**Review by ihhub — APPROVED** [review]

**Comment by ihhub** [comment]
@zenseii , thank you so much for these improvements!

**Pull request merged**